packages/react/src/modules/in-app-message/components/Banner/Banner.tsx

```diff
@@ -160,27 +166,48 @@ const DefaultView: React.FC<{
           />
         )}
 
-        {content.dismissible && (
-          // TODO: Archive message on click
-          <DismissButton />
-        )}
+        {content.dismissible && <DismissButton onClick={onDismissClick} />}
       </Actions>
     </Root>
   );
 };
 
 const Default: React.FC<BannerProps> = ({ filters }) => {
   const { colorMode } = useInAppChannel();
-  const { message } = useInAppMessage<BannerContent>(MESSAGE_TYPE, filters);
+  const { message, inAppMessagesClient } = useInAppMessage<BannerContent>(
+    MESSAGE_TYPE,
+    filters,
+  );
+
+  // Mark the message as seen on render
+  useEffect(() => {
+    if (!message || message.seen_at !== null) return;
+
+    inAppMessagesClient.markAsSeen(message);
+  }, [message, inAppMessagesClient]);
 
   if (!message) return null;
 
-  // TODO: Track interaction on load or whatever other events necessary
+  const onDismissClick = () => {
+    inAppMessagesClient.markAsArchived(message);
+  };
+
+  const onInteracted = () => {
+    inAppMessagesClient.markAsInteracted(message);
+  };
 
-  return <DefaultView content={message.content} colorMode={colorMode} />;
+  return (
+    <DefaultView
+      content={message.content}
+      colorMode={colorMode}
+      onDismissClick={onDismissClick}
+      onInteracted={onInteracted}
```

Comment:
Alternatively,`onDismiss` and `onInteract`? Feels like a few inconsistencies in how they are named.. but naming is hard.

Reply:
Yeah, I like that, it's a bit more consistent

Reply:
Will update this in the modal PR so we can change all of the components